File: my_cpp_pkg/src/activity2_second_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/msg/int64.hpp>

class SecondNode : public rclcpp::Node
{
public:
    SecondNode() : Node("second_node")
    {
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);
        subscriber_ = this->create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&SecondNode::callback_number, this, std::placeholders::_1));

        RCLCPP_INFO(this->get_logger(), "Second Node has been started!");
    }

private:
    int64_t count = 0;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_;

    void callback_number(const example_interfaces::msg::Int64::SharedPtr msg)
    {
        count += msg->data;
        auto msg_pub = example_interfaces::msg::Int64();
        msg_pub.data = count;

        publisher_->publish(msg_pub);
    }
    
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SecondNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: my_py_pkg/my_py_pkg/activity2_second_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64

class SecondNode(Node):
    def __init__(self):
        super().__init__("second_node")

        self.sum = 0

        self.subscriber_ = self.create_subscription(Int64, "number", self.callback_number, 10)
        self.publisher_ = self.create_publisher(Int64, "number_count", 10)
        self.get_logger().info("Second Node has been started!")

    def callback_number(self, msg):
        self.sum += msg.data
        msg_pub = Int64()
        msg_pub.data = self.sum

        self.publisher_.publish(msg_pub)
        # self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    node = SecondNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()